homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,251 @@
+                return await self._validate_and_create_auth(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except AbortFlow:
+                raise
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors or {}
+        )
+
+    async def async_step_user_camera(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[ATTR_SERIAL])
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title=user_input[ATTR_SERIAL],
+                data={
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_TYPE: ATTR_TYPE_CAMERA,
+                },
+            )
+
+        camera_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=DEFAULT_CAMERA_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Required(ATTR_SERIAL): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user_camera", data_schema=camera_schema, errors=errors or {}
+        )
+
+    async def async_step_discovery(self, discovery_info):
+        """Handle a flow for discovered camera without rtsp config entry."""
+
+        await self.async_set_unique_id(discovery_info[ATTR_SERIAL])
+        self._abort_if_unique_id_configured()
+
+        return await self.async_step_confirm(discovery_info)
+
+    async def async_step_confirm(self, confirm_info):
+        """Confirm and create entry from discovery step."""
+        errors = {}
+
+        if confirm_info is not None:
+            if confirm_info.get(CONF_PASSWORD) is not None:
+                return self.async_create_entry(
+                    title=confirm_info[ATTR_SERIAL],
+                    data={
+                        CONF_USERNAME: confirm_info[CONF_USERNAME],
+                        CONF_PASSWORD: confirm_info[CONF_PASSWORD],
+                        CONF_TYPE: ATTR_TYPE_CAMERA,
+                    },
+                )
+
+        discovered_camera_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=DEFAULT_CAMERA_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Required(ATTR_SERIAL, default=self.unique_id): str,
```

Comment:
```suggestion
```

Reply:
How would the user know what camera they are adding credentials to? The integration defaults to the integration name.

That's the reason why I left the serial number field.

![image](https://user-images.githubusercontent.com/66512715/111346293-c09a2800-8686-11eb-8876-aefa1a7e859f.png)

Reply:
We can use title and description place holders with template strings:
https://github.com/home-assistant/core/blob/f3c74948c3a4d2b1048fd980eba9d240db472504/homeassistant/components/deconz/config_flow.py#L209

https://github.com/home-assistant/core/blob/f3c74948c3a4d2b1048fd980eba9d240db472504/homeassistant/components/deconz/strings.json#L3

https://github.com/home-assistant/core/blob/f3c74948c3a4d2b1048fd980eba9d240db472504/homeassistant/components/deconz/config_flow.py#L253

https://github.com/home-assistant/core/blob/f3c74948c3a4d2b1048fd980eba9d240db472504/homeassistant/components/deconz/strings.json#L22

Reply:
Added a description_placeholder. Thank You!